apis/v1alpha2/shared_types.go

```diff
@@ -256,13 +256,65 @@ type RouteStatus struct {
 	// can only populate Route status for the Gateways/parent resources they are
 	// responsible for.
 	//
-	// A maximum of 32 Gateways will be represented in this list. An empty list
-	// means the route has not been attached to any Gateway.
+	// A maximum of 32 parent resources will be represented in this list. An
```

Comment:
We should leave the max number to the kubebuilder validation limit if possible to avoid potential drift between comments and actual value.